Sources/Datadog/Core/System/AppStateListener.swift

```diff
@@ -89,9 +89,16 @@ internal struct AppStateHistory: Equatable {
     }
 }
 
+/// An observer of `AppStateHistory` value.
+internal typealias AppStateHistoryObserver = ValueObserver
```

Comment:
Not sure this has much value 🤔 
The `where` clause in the protocol method is enough to describe the constraint:
```swift
func subscribe<Observer: ValueObserver>(_ subscriber: Observer) where Observer.ObservedValue == AppStateHistory
```

Reply:
This is very small, so I'll keep it for the sake of consistency with other providers like:
https://github.com/DataDog/dd-sdk-ios/blob/bdccc5e652b8f22dd0c18d0d3657fa53c5451898/Sources/Datadog/Core/Attributes/UserInfoProvider.swift#L10
or
https://github.com/DataDog/dd-sdk-ios/blob/bdccc5e652b8f22dd0c18d0d3657fa53c5451898/Sources/Datadog/Core/System/CarrierInfoProvider.swift#L39
but in general I agree - not much value. This will be surely addressed in V2 architecture by getting rid of this by design.